src/mpid/ch4/shm/posix/eager/fbox/fbox_types.h

```diff
@@ -53,7 +53,13 @@ typedef struct MPIDI_POSIX_eager_fbox_control {
     int *local_ranks;
     int *local_procs;
 
-    int next_poll_local_rank;
+    /* A small cache fo local ranks that have posted receives that we use to poll fastboxes more
+     * efficiently. The last entry in this array is a counter to keep track of the most recently
+     * checked fastbox so we can make sure we don't starve the fastboxes where receives haven't been
+     * posted (for unexpected messages). Ideally this array should remain small. By default, it has
+     * three entries for the cache and one entry for the counter, which (at 16 bytes per entry)
+     * stays under a 64 byte cache line size. */
+    int16_t *first_poll_local_ranks;
 
```

Comment:
I do not understand why you use `int16_t` here. Is `int16_t` only 2 bytes  ?

Reply:
RIght. 2 bytes is big enough to hold as many ranks as we're ever likely to have on a node (65536) and it means we can fit the entire array in a cache line if we want to.

Reply:
I added a bit to the comment to help understand this.